akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -607,7 +607,14 @@ class NestedDeferredActorContextSpec extends ActorContextSpec {
   override def decoration[T: ClassTag] = b ⇒ Behaviors.setup(_ ⇒ Behaviors.setup(_ ⇒ b))
 }
 
-class TapActorContextSpec extends ActorContextSpec {
+class InterceptActorContextSpec extends ActorContextSpec {
 
-  override def decoration[T: ClassTag]: Behavior[T] ⇒ Behavior[T] = b ⇒ Behaviors.tap[T](b)((_, _) ⇒ (), (_, _) ⇒ ())
+  def tap[T] = new BehaviorInterceptor[T, T] {
```

Comment:
`val`, to promote the idea that you reuse the same instance in case of nesting

Reply:
Can't do that since `T` is decided by caller through `def decoration[T]` below